test/unit/eigensolver/test_gen_eigensolver.cpp

```diff
@@ -163,21 +165,25 @@ void testGenEigensolver(const blas::Uplo uplo, const SizeType m, const SizeType
   if (mat_a_h.size().isEmpty())
     return;
 
-  testGenEigensolverCorrectness(uplo, reference_a, reference_b, ret, grid...);
+  testGenEigensolverCorrectness(uplo, reference_a, reference_b, ret, 0l, eval_index_end, grid...);
 }
 
 TYPED_TEST(GenEigensolverTestMC, CorrectnessLocal) {
   for (auto uplo : blas_uplos) {
     for (auto [m, mb, b_min] : sizes) {
       getTuneParameters().eigensolver_min_band = b_min;
-      testGenEigensolver<TypeParam, Backend::MC, Device::CPU, Allocation::do_allocation,
-                         Factorization::do_factorization>(uplo, m, mb);
-      testGenEigensolver<TypeParam, Backend::MC, Device::CPU, Allocation::use_preallocated,
-                         Factorization::do_factorization>(uplo, m, mb);
-      testGenEigensolver<TypeParam, Backend::MC, Device::CPU, Allocation::do_allocation,
-                         Factorization::already_factorized>(uplo, m, mb);
-      testGenEigensolver<TypeParam, Backend::MC, Device::CPU, Allocation::use_preallocated,
-                         Factorization::already_factorized>(uplo, m, mb);
+
+      // Test with all evals and half of them
```

Comment:
Is it allowed to ask for 0 eigenvectors/values (or should it at least fail gracefully)? If yes, is it tested?

Reply:
No, currently it fails because of checks on indices access. Thinking about it, I think it should be supported as a "dirty" way to ask only for eigenvalues and no eigenvector. (I say "dirty" because completely removing the eigenvectors would require a separate API for some functions, where you still have to pass in the full eigenvector matrix.) I'm wondering if this should simply be a corner case, where we do a quick return after solving the tridiagonal system?

Reply:
Actually, this limitation was only coming from the correctness test. I'll update the tests accordingly.